overlord/devicestate/devicestate.go

```diff
@@ -393,39 +443,100 @@ func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Mo
 			return nil, err
 		}
 		tss = append(tss, ts)
-	}
-	// gadget
-	if current.Gadget() == new.Gadget() && current.GadgetTrack() != new.GadgetTrack() {
-		ts, err := snapstateUpdateWithDeviceContext(st, new.Gadget(), &snapstate.RevisionOptions{Channel: new.GadgetTrack()}, userID, snapstate.Flags{NoReRefresh: true}, deviceCtx, fromChange)
-		if err != nil {
-			return nil, err
+	} else {
+		var newBaseChannel string
+		if new.Grade() != asserts.ModelGradeUnset && current.Grade() != asserts.ModelGradeUnset {
+			// both old and the new model can specify a default channel
+			if new.BaseSnap().DefaultChannel != current.BaseSnap().DefaultChannel {
+				newBaseChannel = new.BaseSnap().DefaultChannel
+			}
+		} else if new.Grade() != asserts.ModelGradeUnset {
+			// only the new model can specify a default channel,
```

Comment:
this case can't be reached now, because it's a 18->20 remodel? or am I confused?

Reply:
Yes, that would imply 18 to 20 remodel, so it cannot be even tested properly atm.  I can drop this branch and replace it with a TODO in case we actually support this scenario.